src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -412,11 +433,20 @@ void ObjectCopyRequest<I>::compute_src_object_extents() {
     Striper::file_to_extents(m_cct, m_src_image_ctx->format_string,
                              &m_src_image_ctx->layout, e.first, e.second, 0,
                              src_object_extents);
+    auto stripe_unit = std::min(m_src_image_ctx->layout.stripe_unit,
+                                m_dst_image_ctx->layout.stripe_unit);
     for (auto &p : src_object_extents) {
       for (auto &s : p.second) {
         m_src_objects.insert(s.objectno);
-        m_src_object_extents.push_back({s.objectno, s.offset, s.length});
         total += s.length;
+        while (s.length > 0) {
+          assert(s.length >= stripe_unit);
+          auto dst_object_offset = src_to_dst_object_offset(s.objectno, s.offset);
+          m_src_object_extents[dst_object_offset] = {s.objectno, s.offset,
+                                                     stripe_unit};
```

Comment:
What would happen if the ```dst_object_offset``` is in the middle of a dest stripe? For example, if the source extent was 2~2 (stripe of 2) and the destination extext was 0~3 (stripe of 3), the first byte of the source extent would map to the destination extent 2~1 but the last byte of the min stripe would go somewhere else.

Reply:
@dillaman I am not sure I see exactly what you mean, but taking the condition that is used for `stripe_unit` on image creation:
```
(1 << order) % stripe_unit == 0 && stripe_unit <= (1 << order)
```
which I think can be rewritten as:
```
stripe_unit == (1 << x), where x <= order
```
which means that a situation like you described is not possible?

Reply:
OK -- fair enough if odd stripe units are prevented. We have plenty of "fsx" testing time to catch any potential regressions. Go ahead and squash the fixup commit (or just prefix it w/ "librbd") for merging.

Reply:
@dillaman thanks. Rebased.